src/cpp/rtps/transport/shared_mem/SharedMemGlobal.hpp

```diff
@@ -53,10 +54,20 @@ class SharedMemGlobal
             const std::string& domain_name)
         : domain_name_(domain_name)
     {
-        if (domain_name.length() > MAX_DOMAIN_NAME_LENGTH)
+        std::string username;
+        if(fastdds::dds::RETCODE_OK == SystemInfo::get_username(username))
+        {
+            domain_name_ = domain_name_ + "_" + username;
+        }
+        else
+        {
+            throw std::runtime_error("error: couldn't get username to generate port_segment_name");
+        }
+
+        if (domain_name_.length() > MAX_DOMAIN_NAME_LENGTH)
         {
             throw std::runtime_error(
-                      domain_name +
+                      domain_name_ +
```

Comment:
This would most probably fail, since adding the user name will exceed the imposed limit.

Looking at the code, I think we could remove this check if we pass the domain name to the constructor of `SharedMemGlobal::Port`

We will use the domain name passed in, instead of using `node_->domain_name`